src/mainwindow.cpp

```diff
@@ -1519,3 +1520,45 @@ void MainWindow::resetFilters()
     m_hashTableProxy->setShowCheckedRowsOnly(false,false);
     m_hashTableProxy->setFilterRegExp("");
 }
+
+void MainWindow::getDefaultFormat()
+{
+    // This signal may be called 1 ms before applySettings() so use the lineEdit text
+    m_johnDefaultFormat.setJohnProgram(m_ui->lineEditPathToJohn->text());
+    QStringList args;
+    args << "-stdin";
+    args << "--session=defaultFormat";
```

Comment:
What happens if johnDefaultFormat is called at the same time as johnPasswordGuessing or johnAttack or any other JohnAttack derived class if we don't create new session ? According to my knowledge, the default.rec file will be locked for the 2nd process and we'll have problems, so that's why I use a different session. Is that false ?